Userland/Libraries/LibGfx/GIFLoader.cpp

```diff
@@ -705,11 +698,15 @@ ErrorOr<ImageFrameDescriptor> GIFImageDecoderPlugin::frame(size_t index)
         }
     }
 
-    if (m_context->error_state == GIFLoadingContext::ErrorState::NoError && !decode_frame(*m_context, index)) {
-        if (m_context->state < GIFLoadingContext::State::FrameComplete || !decode_frame(*m_context, 0)) {
+    if (m_context->error_state == GIFLoadingContext::ErrorState::NoError && decode_frame(*m_context, index).is_error()) {
+        if (m_context->state < GIFLoadingContext::State::FrameComplete) {
             m_context->error_state = GIFLoadingContext::ErrorState::FailedToDecodeAnyFrame;
             return Error::from_string_literal("GIFImageDecoderPlugin: Decoding failed");
         }
+        if (auto result = decode_frame(*m_context, 0); result.is_error()) {
+            m_context->error_state = GIFLoadingContext::ErrorState::FailedToDecodeAnyFrame;
+            return result.release_error();
+        }
```

Comment:
Ok. I did not move `m_context->error_state = GIFLoadingContext::ErrorState::FailedToDecodeAnyFrame;` before the last call to `decode_frame(*m_context, 0)` because `decode_frame` could modify `m_context->error_state`